{"version":3,"sources":["../../app-src/util/ApplicationException.js"],"names":["ApplicationException","Error","constructor","msg","name","exception","isApplicationException","error","Object","getPrototypeOf","getExceptionMessage","message","console","log"],"mappings":";;;;;;AAAO,YAAMA,oBAAN,SAAmCC,KAAnC,CAAyC;AAC9CC,oBAAYC,MAAM,EAAlB,EAAsB;AACpB,gBAAMA,GAAN;AACA,eAAKC,IAAL,GAAY,KAAKF,WAAL,CAAiBE,IAA7B;AACD;AAJ6C;;;;AAOhD,YAAMC,YAAYL,oBAAlB;;AAEO,eAASM,sBAAT,CAAgCC,KAAhC,EAAuC;AAC5C,eACEA,iBAAiBF,SAAjB,IACAG,OAAOC,cAAP,CAAsBF,KAAtB,aAAwCF,SAF1C;AAID;;;;AAEM,eAASK,mBAAT,CAA6BH,KAA7B,EAAoC;AACzC,YAAID,uBAAuBC,KAAvB,CAAJ,EAAmC;AACjC,iBAAOA,MAAMI,OAAb;AACD,SAFD,MAEO;AACLC,kBAAQC,GAAR,CAAYN,KAAZ;AACA,iBAAO,uCAAP;AACD;AACF","file":"ApplicationException.js","sourcesContent":["export class ApplicationException extends Error {\n  constructor(msg = \"\") {\n    super(msg);\n    this.name = this.constructor.name;\n  }\n}\n\nconst exception = ApplicationException;\n\nexport function isApplicationException(error) {\n  return (\n    error instanceof exception ||\n    Object.getPrototypeOf(error) instanceof exception\n  );\n}\n\nexport function getExceptionMessage(error) {\n  if (isApplicationException(error)) {\n    return error.message;\n  } else {\n    console.log(error);\n    return \"Não foi possível realizar a operação.\";\n  }\n}\n"]}